### PR TITLE
RD-233 - Add customization support for Composer

### PR DESCRIPTION
**Pull request by qooban**
### Changes
* Removed Cloudify word from Composer section
* Added documentation page for Composer customization
* Minor change in overrides.md

Note 1: In case of Composer we are not able to provide link to GitHub repository (eg. for the base translation file) as this repository is private (so only accessible by Cloudify). I created [RD-637](https://cloudifysource.atlassian.net/browse/RD-637) for that. 

Note 2: There are some alignments (between Stage and Composer) which we should consider. I added [RD-635](https://cloudifysource.atlassian.net/browse/RD-635) for small tasks, but generally [RD-605](https://cloudifysource.atlassian.net/browse/RD-605) needs to be reviewed and planned.

### CODE REVIEW

File: content/developer/composer/customization.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            "logoUrl": "/composer/backend/userData/static/logo.png",
+            "mainColor": "grey"
+        }
+    }
+```
+
+Default configuration file is available at
+`/opt/cloudify-composer/backend/conf/userConfig.json`.
+
+All parameters description is available in
```
Comment (kubama) [inline content/developer/composer/customization.md:50]:
```suggestion
All configuration parameters are described in
```

File: content/developer/composer/customization.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+1. An example file content may look like:
+```json
+    {
+        "theme": {
+            "logoUrl": "/composer/backend/userData/static/logo.png",
+            "mainColor": "grey"
+        }
+    }
+```
```
[the remark this answers is not in the document]
Comment (qooban) [reply content/developer/composer/customization.md]: Fixed.

File: content/developer/composer/customization.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+* to see the applied changes you need to restart {{< param cfy_composer_name >}} 
+service on the {{< param cfy_manager_name >}} machine by executing: 
+```shell script
+sudo service cloudify-composer.service restart
+```
+* in {{< param cfy_manager_name >}} Cluster environment, 
+these changes should be applied to every {{< param cfy_manager_name >}} in the cluster
+
+
+## Theme setup
```
Comment (kubama) [inline content/developer/composer/customization.md:29]: Order in which information is given in this section is counter intuitive. What's in lines 49-53 is vital and should be placed immediately after line 35. It should be also explicitly stated what should be placed in the created file - the sentence 'Default configuration file is available at ...' doesn't really tell how created file is related to this default configuration file (overriding).
Example is least important and should be moved to the end.

Comment (qooban) [reply content/developer/composer/customization.md:29]: Changed it a bit.

**Review by kubama — APPROVED** [review]

File: content/developer/composer/customization.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+layout: bt_wiki
+title: Customization
+category: Composer
+draft: false
+weight: 600
+userdata_path: "/opt/cloudify-composer/backend/dev"
+config_path: "/opt/cloudify-composer/backend/conf"
+---
+
+{{< param cfy_composer_name >}} application provides basic support for customizing the layout.
```
Comment (ofercloudify) [inline content/developer/composer/customization.md:11]:
```suggestion
Similar to the {{< param cfy_console_name >}} white labeling options, the {{< param cfy_composer_name >}} provides basic layout customization options.
```

File: content/developer/composer/customization.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+Main features are:
```
Comment (ofercloudify) [inline content/developer/composer/customization.md:12]:
```suggestion
The main supported features are:
```

File: content/developer/composer/customization.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+* adding custom logo and setting main application color
```
Comment (ofercloudify) [inline content/developer/composer/customization.md:14]:
```suggestion
* Setting a custom logo
* Setting the main application color
```

File: content/developer/composer/customization.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+* styling the HTML elements
```
Comment (ofercloudify) [inline content/developer/composer/customization.md:15]:
```suggestion
* Styling the HTML elements
```

File: content/developer/composer/customization.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+* changing text labels
```
Comment (ofercloudify) [inline content/developer/composer/customization.md:16]:
```suggestion
* Changing text labels 
```

File: content/developer/composer/customization.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+---
+
+{{< param cfy_composer_name >}} application provides basic support for customizing the layout.
+Main features are:
+
+* adding custom logo and setting main application color
+* styling the HTML elements
+* changing text labels 
+
+When introducing changes described in this page you should be aware that:
```
Comment (ofercloudify) [inline content/developer/composer/customization.md:18]:
```suggestion
When introducing the changes described on this page, please note:
```

File: content/developer/composer/customization.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+{{< param cfy_composer_name >}} application provides basic support for customizing the layout.
+Main features are:
+
+* adding custom logo and setting main application color
+* styling the HTML elements
+* changing text labels 
+
+When introducing changes described in this page you should be aware that: 
+
+* to see the applied changes you need to restart {{< param cfy_composer_name >}}
```
Comment (ofercloudify) [inline content/developer/composer/customization.md:20]:
```suggestion
* To see the applied changes you need to restart the {{< param cfy_composer_name >}} 
```

File: content/developer/composer/customization.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+* changing text labels 
+
+When introducing changes described in this page you should be aware that: 
+
+* to see the applied changes you need to restart {{< param cfy_composer_name >}} 
+service on the {{< param cfy_manager_name >}} machine by executing: 
+```shell script
+sudo service cloudify-composer.service restart
+```
+* in {{< param cfy_manager_name >}} Cluster environment,
```
Comment (ofercloudify) [inline content/developer/composer/customization.md:25]:
```suggestion
* In a {{< param cfy_manager_name >}} Cluster environment, 
```

File: content/developer/composer/customization.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+When introducing changes described in this page you should be aware that: 
+
+* to see the applied changes you need to restart {{< param cfy_composer_name >}} 
+service on the {{< param cfy_manager_name >}} machine by executing: 
+```shell script
+sudo service cloudify-composer.service restart
+```
+* in {{< param cfy_manager_name >}} Cluster environment, 
+these changes should be applied to every {{< param cfy_manager_name >}} in the cluster
```
Comment (ofercloudify) [inline content/developer/composer/customization.md:26]:
```suggestion
these changes should be applied to every {{< param cfy_manager_name >}} in the cluster.
```

File: content/developer/composer/customization.md [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+
+## Theme setup
+
+{{< param cfy_composer_name >}} provides a way to introduce basic theme changes 
+(eg. logo URL, main color, header text color) through user configuration file 
+residing on the {{< param cfy_manager_name >}}.
+
+To setup theme create `userConfig.json` file in `{{< field "userdata_path" >}}` directory.
```
Comment (ofercloudify) [inline content/developer/composer/customization.md:35]:
```suggestion
To setup a theme, create `userConfig.json` file in the `{{< field "userdata_path" >}}` directory.
```

**Review by ofercloudify — CHANGES_REQUESTED** [review]

**Review by ofercloudify — APPROVED** [review]

**Pull request merged**